Homework/homework6/task1.py

```diff
@@ -0,0 +1,110 @@
+"""
+В этом задании будем улучшать нашу систему классов из задания прошлой лекции.
+
+Student, Teacher, Homework
+Советую обратить внимание на defaultdict из модуля collection для
+использования как общую переменную
+1. Как то не правильно, что после do_homework мы возвращаем все тот же
+объект - будем возвращать какой-то результат работы (HomeworkResult)
+HomeworkResult принимает объект автора задания, принимает исходное задание
+и его решение в виде строки
+Атрибуты:
+    homework - для объекта Homework, если передан не этот класс -  выкинуть
+    подходящие по смыслу исключение с сообщением:
+    'You gave a not Homework object'
+    solution - хранит решение ДЗ как строку
+    author - хранит объект Student
+    created - c точной датой и временем создания
+2. Если задание уже просрочено хотелось бы видеть исключение при do_homework,
+а не просто принт 'You are late'.
+Поднимайте исключение DeadlineError с сообщением 'You are late' вместо print.
+3. Student и Teacher имеют одинаковые по смыслу атрибуты
+(last_name, first_name) - избавиться от дублирования с помощью наследования
+4.
+Teacher
+Атрибут:
+    homework_done - структура с интерфейсом как в словаря, сюда поподают все
+    HomeworkResult после успешного прохождения check_homework
+    (нужно гаранитровать остутствие повторяющихся результатов по каждому
+    заданию), группировать по экземплярам Homework.
+    Общий для всех учителей. Вариант ипользования смотри в блоке if __main__...
+Методы:
+    check_homework - принимает экземпляр HomeworkResult и возвращает True если
+    ответ студента больше 5 символов, так же при успешной проверке добавить в
+    homework_done.
+    Если меньше 5 символов - никуда не добавлять и вернуть False.
+    reset_results - если передать экземпряр Homework - удаляет только
+    результаты этого задания из homework_done, если ничего не передавать,
+    то полностью обнулит homework_done.
+PEP8 соблюдать строго.
+Всем перечисленным выше атрибутам и методам классов сохранить названия.
+К названием остальных переменных, классов и тд. подходить ответственно -
+давать логичные подходящие имена.
+"""
+import datetime
+from collections import defaultdict
+
+
+class MyLibraryError(Exception):
```

Comment:
It is used only in a single place.  Please remove. https://ru.wikipedia.org/wiki/YAGNI

Reply:
Okay, I remember from the lecture that its a kind of the best practice, so I thought maybe it also should be if we have only one created-by-me exception class.